core/executor/src/wasmi_execution.rs

```diff
@@ -568,9 +551,8 @@ impl WasmRuntime for WasmiRuntime {
 		method: &str,
 		data: &[u8],
 	) -> Result<Vec<u8>, Error> {
-		self.with(|module| {
-			call_in_wasm_module(ext, module, method, data, &self.host_functions)
-		})
+		self.state_snapshot.apply(&self.instance)?;
```

Comment:
This will fail the execution. Why not reset the instance?

Reply:
What do you mean by reset here?

If you mean call `invalidate_runtime`, then I'd expect it to happen as part of handling `Err`.

Reply:
That does not invalidates the runtime. I mean reset `instance` and `module` of your current runtime and retry the execution.

The problem when a call fails, it could invalidate a transaction when this happens while building a block or when importing one.